spinoso-string/src/lib.rs

```diff
@@ -362,8 +157,9 @@ impl String {
     #[must_use]
     pub fn with_capacity(capacity: usize) -> Self {
         let buf = Vec::with_capacity(capacity);
-        let encoding = Encoding::Utf8;
-        Self { buf, encoding }
+        Self {
+            inner: EncodedString::new(buf, Encoding::Utf8),
```

Comment:
As a nit, I'd rather add a `EncodedString::utf8` constructor, since we pay to branch on the encoding in `EncodedString::new` when we don't need to.

Reply:
Filed:

- https://github.com/artichoke/artichoke/issues/1725